### PR TITLE
upgrade(installer): Let installer E2E use packages built in the CI

### PR DESCRIPTION
**Pull request by BaptisteFoy**
### What does this PR do?
Push a by-pipeline version of the installer & agent OCIs to the QA registry to be used in E2E VMs

### Motivation
Using what's built in a pipeline in E2E tests

### Additional Notes
N/A

### Possible Drawbacks / Trade-offs
N/A

### Describe how to test/QA your changes
Make sure that E2E works or setup a test VM and check that the installer OCI version is the one of your pipeline

### CODE REVIEW

File: .gitlab/e2e/e2e.yml [hunk 1]
```diff
@@ -288,6 +288,8 @@ new-e2e-installer:
     - deploy_deb_testing-i7_amd64
     - deploy_rpm_testing-i7_arm64
     - deploy_rpm_testing-i7_x64
+    - deploy_installer_oci
+    - deploy_agent_oci
```
Comment (BaptisteFoy) [inline .gitlab/e2e/e2e.yml:292]: Not used now but will be useful when testing the download of the commit's agent OCI

**Review by coignetp — APPROVED** [review]

**Review by coignetp — APPROVED** [review]

File: omnibus/package-scripts/installer-deb/postinst [hunk 2]
```diff
@@ -11,8 +11,13 @@ readonly STABLE_INSTALLER=${PACKAGES_DIR}/datadog-installer/stable/bin/installer
 readonly PACKAGES_LOCK_DIR=/var/run/datadog-packages
 
 bootstrap() {
-    $BOOTSTRAP_INSTALLER bootstrap --package "datadog-installer" && \
-    $STABLE_INSTALLER bootstrap
+    if [ -z "$DD_INSTALLER_OCI_BOOTSTRAP_VERSION" ]; then
```
Comment (raphaelgavache) [inline omnibus/package-scripts/installer-deb/postinst:14]: this is removed here https://github.com/DataDog/datadog-agent/pull/24920#pullrequestreview-2014970530

Comment (BaptisteFoy) [reply omnibus/package-scripts/installer-deb/postinst:14]: will update once merged

**Review by raphaelgavache — APPROVED** [review]

**Review by KevinFairise2 — APPROVED** [review]

**Review by chouquette — APPROVED** [review]

**Review by clamoriniere — APPROVED** [review]
/lgtm

**Pull request merged**